src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -54,113 +54,116 @@ void *AlignMem(void *mem, size_t size, size_t alignment, size_t *space) {
   return reinterpret_cast<void *>(addr);
 }
 
-mkldnn::memory *TmpMemMgr::Alloc(const mkldnn::memory::primitive_desc &pd) {
+mkldnn::memory *TmpMemMgr::Alloc(const mkldnn::memory::desc &md) {
   // We need to include the size of the memory used for alignment.
-  this->est_size += pd.get_size() + alignment;
-  void *mem = AlignMem(this->curr_mem, pd.get_size(), alignment, &this->curr_size);
+  this->est_size += md.get_size() + alignment;
+  void *mem = AlignMem(this->curr_mem, md.get_size(), alignment, &this->curr_size);
   if (mem) {
     // The memory is allocated from the temporary memory space in the
     // operator. It'll only become invalid after we exit from the operator.
-    mkldnn_mem_ptr ret(new mkldnn::memory(pd, mem));
+    mkldnn_mem_ptr ret(new mkldnn::memory(md, CpuEngine::Get()->get_engine(), mem));
     MKLDNNStream::Get()->RegisterMem(ret);
     CHECK_EQ(mem, mem);
-    this->curr_size -= pd.get_size();
-    this->curr_mem = static_cast<char *>(mem) + pd.get_size();
+    this->curr_size -= md.get_size();
+    this->curr_mem = static_cast<char *>(mem) + md.get_size();
     return ret.get();
   } else {
     // If curr_mem has been initialized and we still reach here. It means
     // the current allocated memory isn't enough.
     if (this->curr_mem && dmlc::GetEnv("MXNET_MKLDNN_DEBUG", false)) {
-      LOG(WARNING) << "Allocate " << pd.get_size()
+      LOG(WARNING) << "Allocate " << md.get_size()
```

Comment:
Actually, it doesn't matter that the currently allocated memory from MXNet resources isn't enough. MKL-DNN could allocate the required space by itself. Thus, we just let it continue and give the message for the developers (controlled by `MXNET_MKLDNN_DEBUG`). Here we use `TmpMemMgr` to estimate the maximum space size. As to the `else` branch above, it would be reached for once or twice. Then `TmpMemMgr` would allocate the space with the maximum size. We will add some annotations and log messages using another PR. Thanks.

Reply:
Thanks for the explanation